monitor/puller.go

```diff
@@ -101,6 +103,28 @@ func (m *Monitor) syncIntendWithdrawChannel(intendWithdrawChannel *mainchain.Cel
 	}
 }
 
+func (m *Monitor) syncSGNAddSubscriptionBalance(event *mainchain.SGNAddSubscriptionBalance) {
+	transactor := m.Transactor
+	consumer := event.Consumer
+	consumerEthAddress := consumer.Hex()
+	amount := event.Amount
+	amountInt := sdk.NewIntFromBigInt(amount)
+	subscription, err := guard.CLIQuerySubscription(transactor.CliCtx, guard.RouterKey, consumerEthAddress)
+	if err == nil {
+		if subscription.Deposit.Equal(amountInt) {
+			log.Infof("Subscription already updated for %s, amount %s", consumerEthAddress, amount)
+			return
+		}
+	}
+	log.Infof("Sync AddSubscriptionBalance for %s, amount %s, tx hash %x", consumerEthAddress, amount, event.Raw.TxHash)
+	subscription = guard.NewSubscription(consumer.Hex())
+	subscription.Deposit = sdk.NewIntFromBigInt(amount)
+	subscriptionData := transactor.CliCtx.Codec.MustMarshalBinaryBare(subscription)
+	msg := sync.NewMsgSubmitChange(sync.Subscribe, subscriptionData, m.Transactor.Key.GetAddress())
+	log.Infof("submit change tx: subscribe EthAddress %s, Amount %s", consumer, amount)
```

Comment:
nit: combine logs of line 119 and 124 to one log line, as they have mostly duplicated info

Reply:
Done.